test/Microsoft.AspNetCore.Mvc.Core.Test/Infrastructure/DefaultAssemblyProviderTests.cs

```diff
@@ -0,0 +1,216 @@
+        {
+            // Arrange
+            var dependencyContext = new DependencyContext(
+               target: null,
+               runtime: null,
+               compilationOptions: CompilationOptions.Default,
+               compileLibraries: new CompilationLibrary[0],
+               runtimeLibraries: new[]
+               {
+                     GetLibrary("Foo", "CustomMvc.Modules"),
+                     GetLibrary("Bar", "CustomMvc.Application.Loader"),
+                     GetLibrary("Baz", "Microsoft.AspNetCore.Mvc.Abstractions"),
+               });
+            var assemblyProvider = new OverridenAssemblyProvider(
+                CurrentAssembly,
+                dependencyContext,
+                referenceAssemblies: null);
+
+            // Act
+            var candidates = assemblyProvider.GetCandidateLibraries();
+
+            // Assert
+            Assert.Empty(candidates);
+        }
+
+        // This test verifies DefaultAssemblyProvider.ReferenceAssemblies reflects the actual loadable assemblies
+        // of the libraries that Microsoft.AspNetCore.Mvc dependes on.
+        // If we add or remove dependencies, this test should be changed together.
+        [Fact]
+        public void ReferenceAssemblies_ReturnsLoadableReferenceAssemblies()
+        {
+            // Arrange
+            var provider = new TestableAssemblyProvider(CurrentAssembly, dependencyContext: null);
+            var exceptionalAssemblies = new string[]
+            {
+                "Microsoft.AspNetCore.Mvc.WebApiCompatShim",
+            };
+
+            var additionalAssemblies = new[]
+            {
+                // The following assemblies are not reachable from Microsoft.AspNetCore.Mvc
+                "Microsoft.AspNetCore.Mvc.TagHelpers",
+                "Microsoft.AspNetCore.Mvc.Formatters.Xml",
+            };
+
+            var expected = DependencyContext.Load(typeof(MvcServiceCollectionExtensions).GetTypeInfo().Assembly)
+                .RuntimeLibraries
+                .Where(r => r.PackageName.StartsWith("Microsoft.AspNetCore.Mvc", StringComparison.Ordinal) &&
+                    !exceptionalAssemblies.Contains(r.PackageName, StringComparer.OrdinalIgnoreCase))
+                .Select(r => r.PackageName)
+                .Concat(additionalAssemblies)
+                .OrderBy(p => p, StringComparer.Ordinal);
+
+            // Act
+            var referenceAssemblies = provider.ReferenceAssemblies.OrderBy(p => p, StringComparer.Ordinal);
+
+            // Assert
+            Assert.Equal(expected, referenceAssemblies);
+        }
+
+        private static RuntimeLibrary GetLibrary(string name, params string[] dependencyNames)
+        {
+            var dependencies = dependencyNames?.Select(d => new Dependency(d, "1.0.0")) ?? new Dependency[0];
+
+            return new RuntimeLibrary(
+                "package",
+                name,
+                "1.0.0",
```

Comment:
Anyway we can grab the version in a way that wont require code updates every time we ship?